Smart Search - move Filters - v2 to correct merge conflict

pe7er opened the pull request:
New PR to solve Merge Conflict of PR #6959 "Smart Search - move Filters (3x) from left to [Search Tools] "

Please follow the testing procedure of PR #6959

---

This PR moves the Filter option of all three views of **Smart Search** to the [Search Tools] button.
(See Issue #6941 regarding the inconsistency with Filter options).
## Test instructions

You have to enable the **Content - Smart Search** Plugin, and **Index your test website**
Please check the 3 different views before & after the patch: 
test all the filters and search box, clear results, choose max amount of results on a page.
## Smart Search: Manage Indexed Content
### Check the current position of the Filters

In back-end > Components > Smart Search > Indexed Content
The Filters are on the left hand side

![screen shot 2015-05-15 at 14 22 52](http://issues.joomla.org/uploads/1/9b6cf2caa2a5ca839c81409974b03437.png)
### This PR changes moves the Filter

from the left to the middle column and will be visible when you click on the [Search Tools] button.

![screen shot 2015-05-15 at 14 22 52](http://issues.joomla.org/uploads/1/741fb1b5b331b553bdbfc926edef7195.png)
## Smart Search: Manage Content Maps
### Check the current position of the Filters

In back-end > Components > Smart Search > Content Maps
The Filters are on the left hand side

![screen shot 2015-05-15 at 14 22 52](http://issues.joomla.org/uploads/1/7f31ad619fce230e4e1aaf825d1847f0.png)
### This PR changes moves the Filter

from the left to the middle column and will be visible when you click on the [Search Tools] button.

![screen shot 2015-05-15 at 14 22 52](http://issues.joomla.org/uploads/1/f2ed1d2e47e52ba1af0f66195a957e5f.png)
## Smart Search: Manage Search Filters
### Check the current position of the Filters

In back-end > Components > Smart Search > Search Filters
The Filters are on the left hand side

![screen shot 2015-05-15 at 14 22 52](http://issues.joomla.org/uploads/1/a2f3a221a31b8072071682887eabcefb.png)
### This PR changes moves the Filter

from the left to the middle column and will be visible when you click on the [Search Tools] button.

![screen shot 2015-05-15 at 14 22 52](http://issues.joomla.org/uploads/1/5314254c1b4b1a82f10485c12d4bff8d.png)
## Note

To Clear the Filters / Search box, you can use the [Clear] button.
However, sometimes you have to press 2 times to clear the results.
I don't know the cause and how to solve it...

Commit:
Smart Search - move Filters - v2 to correct merge conflict

jduerscheid commented:
I have tested this item :white_check_mark: successfully on 9a763fbba7551d868abb60e11a66faac3942f750<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8140">issues.joomla.org/joomla-cms/8140</a>.</sub>

roland-d commented on administrator/components/com_finder/views/index/tmpl/default.php:
```diff
@@ -60,19 +60,7 @@
-			</div>
-		</div>
+		<?php echo JLayoutHelper::render('joomla.searchtools.default', array('view' => $this)); ?>
 		<div class="clearfix"> </div>
```
@pe7er I think the clearfix should be removed here. You did that in the old PR. Can you check please?

roland-d commented on administrator/components/com_finder/views/maps/tmpl/default.php:
```diff
@@ -46,15 +46,7 @@
-			</div>
-		</div>
+		<?php echo JLayoutHelper::render('joomla.searchtools.default', array('view' => $this)); ?>
 		<div class="clearfix"> </div>
```
@pe7er The clearfix should be mroeved I think as you did in your old PR. Can you check please?

roland-d commented on administrator/components/com_finder/views/maps/view.html.php:
```diff
@@ -92,20 +94,5 @@ protected function addToolbar()
 			JToolbarHelper::divider();
 		}
-
-		JHtmlSidebar::setAction('index.php?option=com_finder&view=maps');
```
@pe7er In your old PR this line is still there, should it remain or be removed?

Commit:
removed clearfix

pe7er commented:
@roland-d I think that JHtmlSidebar::setAction can be removed.
I've removed the clearfix in all three list views.

n9iels commented:
I applied the patch before activating the content plugin. The message that tells me that is now very close to the search filters. Maybe we can add some padding there?
![smart search message](https://cloud.githubusercontent.com/assets/2659866/10866065/4277af86-8021-11e5-8f96-eefef7595fea.PNG)

I also noticed that the filter on "Content maps" are always visible, and for some reason the filter on the right is always marked as an active filter (There are blue lines around it).
On other places where we use this filters (com_content for example) the blue line only appears when we choose an option of the filter.

Commit:
Merge branch 'staging' of https://github.com/joomla/joomla-cms into search-tools-filters-smartsearch2

Commit:
corrected Select Branch filter

pe7er commented:
Thanks @n9iels, I corrected the Branch filter on "Content maps".
Now it is not activated automatically anymore.

btw: I did not change the padding of the warning message...

n9iels commented:
Yes, that works for me! :smile:  I see now the same problem appears for "indexed content", the search filter "Select type of content" is always active over there.

Also the search filter on "Content mapping" are always visible on that page. Or is that done by a certain reason?

n9iels commented:
About that padding and the message. Strange enough the message appears below the search filters. Other messages like "save" and "delete" appear above the messages.

Seems like all other message than "successfully saved", "successfully deleted" etc. appear below the filters. Not a problem for this PR I think so.

pe7er commented:
Thanks for testing! 

No, I did not set the filter to "always active" on purpose. I'll check if I can correct it.

If someone knows the cause of the problem, please correct it with a PR, 
or tell me what to change & I'll do so...

roland-d commented:
@pe7er The "always active" happens because you have set a default value of 0 here: https://github.com/joomla/joomla-cms/pull/8140/files#diff-762a4c09ff9b1a26880f1391232575c2R23

The first option you give to the dropdown has a value of 0:
https://github.com/joomla/joomla-cms/pull/8140/files#diff-762a4c09ff9b1a26880f1391232575c2R29

that is why the first entry becomes an active one. I guess you can drop the default value setting.

Commit:
Merge branch 'staging' of https://github.com/joomla/joomla-cms into search-tools-filters-smartsearch2

Commit:
changed default option value of filter_index.xml

Commit:
fixed Select Branch dropdown in Smart Search Content Maps

zero-24 commented:
@pe7er Travis cry

```
FILE: ...d/joomla/joomla-cms/administrator/components/com_finder/models/maps.php
--------------------------------------------------------------------------------
FOUND 2 ERROR(S) AFFECTING 2 LINE(S)
--------------------------------------------------------------------------------
 176 | ERROR | Expected "if (...)\n...{...}\n...else\n"; found "if
     |       | (...)\n...{...}\n...else"
 184 | ERROR | Expected "if (...)\n...{...}\n...else\n"; found "if
     |       | (...)\n...{...}...else"
--------------------------------------------------------------------------------
UPGRADE TO PHP_CODESNIFFER 2.0 TO FIX ERRORS AUTOMATICALLY
--------------------------------------------------------------------------------
```

roland-d commented:
I don't think the last change is correct. In Joomla 3.4.5 the dropdown looks like this:
![finder](https://cloud.githubusercontent.com/assets/359377/10945146/d5f19e4a-831c-11e5-8d58-5e26164066c3.png)

Branches only as first option and I think we should keep that.

Commit:
undoing: fixed Select Branch dropdown in Smart Search Content Maps

pe7er commented:
I removed my previous PR "fixed Select Branch dropdown in Smart Search Content Maps"

Kubik-Rubik commented:
I have tested this item :white_check_mark: successfully on 26670dd3f5580b7f6426e4ea01942af31ba48dea<br /><br />Works as described. Thank you @pe7er!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8140">issues.joomla.org/joomla-cms/8140</a>.</sub>

Kubik-Rubik commented:
Works but a small design problem:

![2015-11-04 21_32_38-joomla-cms - administration - smart search_ indexed content](https://cloud.githubusercontent.com/assets/1976103/10951108/d86d5a50-833b-11e5-9694-dafaaafdd1e8.png)

This can also be fixed after we've merged this PR!

dgt41 commented:
I have tested this item :white_check_mark: successfully on 26670dd3f5580b7f6426e4ea01942af31ba48dea<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8140">issues.joomla.org/joomla-cms/8140</a>.</sub>

dgt41 commented:
To solve the problem Victor mentioned above just move the filters under the message
like this:
```php
		<?php if (!$this->pluginState['plg_content_finder']->enabled) : ?>
			<div class="alert fade in">
				<button class="close" data-dismiss="alert">×</button>
				<?php echo JText::_('COM_FINDER_INDEX_PLUGIN_CONTENT_NOT_ENABLED'); ?>
			</div>
		<?php endif; ?>
		<?php echo JLayoutHelper::render('joomla.searchtools.default', array('view' => $this)); ?>
```

n9iels commented:
I spotted that design problem to. Maybe it is an idea to move that message to a `JFactory::getApplication()->enqueueMessage`? There are more component with that problem, where the search filters will be add.

zero-24 commented:
RTC lets merge here and @n9iels or @dgt41 can send a fix for the design problem. Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8140">issues.joomla.org/joomla-cms/8140</a>.</sub>

roland-d commented:
The branches dropdown needs to be checked. It is quite different from the current implementation.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8140">issues.joomla.org/joomla-cms/8140</a>.</sub>

zero-24 commented:
Ah sorry overread this comment ;)

dgt41 commented:
Looks good to me:
![screenshot 2015-11-04 23 36 38](https://cloud.githubusercontent.com/assets/3889375/10952749/0cd988de-834d-11e5-9a5d-90fd381cc534.png)

roland-d commented:
@dgt41 Compare it to the implementation in J3.4.5, it is quite different. The current implementation uses it's own JHtml class, that is completely skipped now.

dgt41 commented:
ahh yeah, the finder way 😝

Commit:
Merge branch 'staging' of https://github.com/joomla/joomla-cms into search-tools-filters-smartsearch2

Commit:
branches now retrieved via fields

pe7er commented:
I've removed the hardcoded SQL
query="SELECT id AS value, title AS branch FROM #__finder_taxonomy WHERE parent_id = 1"
from /administrator/components/com_finder/models/forms/filter_maps.xml and
added /administrator/components/com_finder/models/fields/branches.php 
to retrieve the branches.

roland-d commented on administrator/components/com_finder/models/maps.php:
```diff
@@ -263,7 +264,7 @@ protected function populateState($ordering = null, $direction = null)
 		$this->setState('filter.state', $state);
 
-		$branch = $this->getUserStateFromRequest($this->context . '.filter.branch', 'filter_branch', '1', 'string');
+		$branch = $this->getUserStateFromRequest($this->context . '.filter.branch', 'filter_branch', '', 'string');
```
Don't we have to put back the default value here?

dgt41 commented:
@pe7er I am afraid that administrator/components/com_finder/models/fields/branches.php doesn;t cover all the functionality of the old JHtml class https://github.com/joomla/joomla-cms/blob/staging/components/com_finder/helpers/html/filter.php#L208-L394

roland-d commented on administrator/components/com_finder/models/fields/branches.php:
```diff
@@ -0,0 +1,52 @@
+	public function getOptions()
+	{
+		// Build the query.
+		$db = JFactory::getDbo();
```
@pe7er Just change this complete function to contain:
```
return JHtml::_('finder.mapslist');

Commit:
removed SQL query from fields/branches.php

pe7er commented:
Removed the SQL query from fields/branches.php & replaced with return JHtml::_('finder.mapslist');
Thanks for the tip @roland-d !

dgt41 commented:
@pe7er can you also implement this https://github.com/joomla/joomla-cms/pull/8140#issuecomment-153859716
so I can re-test this?

pe7er commented:
@dgt41 yes, please! 
I think that we should correct the layout issue with the message later because it might need to be corrected at other places as well...

dgt41 commented:
Works fine here, one small UX tho:
![screenshot 2015-11-05 01 18 30](https://cloud.githubusercontent.com/assets/3889375/10955312/a2c679b6-835b-11e5-86bb-aa43be6e5b31.png)

designbengel commented:
Small issue:

Before:
![bildschirmfoto 2015-11-05 um 00 22 00](https://cloud.githubusercontent.com/assets/828371/10955328/5b55ca80-8353-11e5-87d8-bdce16aabc5a.png)

After patch:
![bildschirmfoto 2015-11-05 um 00 22 41](https://cloud.githubusercontent.com/assets/828371/10955329/5d574b10-8353-11e5-855c-da2dd8c26cde.png)

designbengel commented:
I don´t know if this regards this PR but if i don´t have published filters there is a message that "No filters have been created yet.	Create a filter." - that´s not true because there are filters but they are unpublished. 
![bildschirmfoto 2015-11-05 um 00 25 04](https://cloud.githubusercontent.com/assets/828371/10955410/d754ad0e-8353-11e5-889d-6adb04cc216b.png)

pe7er commented:
Thanks for testing @dgt41 & @designbengel !
Those UX issues need to be corrected, but I think that it could be done for beta2 in another PR.

designbengel commented:
There is a Message that no content has been indexed by filtering unpublished items?

![bildschirmfoto 2015-11-05 um 00 27 24](https://cloud.githubusercontent.com/assets/828371/10955438/0881ff80-8354-11e5-81f0-1502d3d93b2d.png)

roland-d commented:
@designbengel I think this is an existing issue, this PR only moves the buttons from the sidebur to the Search Tools bar. Perhaps raise a separate issue for that?

designbengel commented:
I have tested this item :white_check_mark: successfully on b14cbd40ec4583a94d81a0d7386c7b024314249c<br /><br />Ok, so like @roland-d says it´s regarding just the location-change of filters - so test for this is ok :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8140">issues.joomla.org/joomla-cms/8140</a>.</sub>

infograf768 commented:
Folks, I can't see
COM_FINDER_MAPS_SELECT_BRANCHE="- Select Branch -"
anywhere.

![screen shot 2015-11-27 at 17 43 51](https://cloud.githubusercontent.com/assets/869724/11445676/86eab6aa-952e-11e5-8d91-52c127601df7.png)

dgt41 commented:
This is normal, you have to be INSIDE a branch for the option to appear (in fact this is a way to go one level back)
![screen shot 2015-11-27 at 18 48 56](https://cloud.githubusercontent.com/assets/3889375/11445775/bf515a7c-9537-11e5-86b9-501556626b40.png)

infograf768 commented:
@pe7er  @roland-d 

I have the feeling some new lang strings were merged here that should'nt have...

infograf768 commented:
@pe7er @roland-d Please test
https://github.com/joomla/joomla-cms/issues/8558
